airbyte-integrations/connectors/source-harvest/source_harvest/streams.py

```diff
@@ -0,0 +1,296 @@
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field, [])
+        yield from records
+
+
+class HarvestStreamIncrementalMixin(HttpStream, ABC):
+    cursor_field = "updated_at"
+
+    def __init__(self, updated_since: pendulum.datetime, **kwargs):
+        super().__init__(**kwargs)
+        self._updated_since = updated_since
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        updated_since = self._updated_since
+        if stream_state.get(self.cursor_field):
+            updated_since = stream_state[self.cursor_field]
+        params.update({"updated_since": updated_since})
+        return params
+
+
+class HarvestStreamWithPaginationSliced(HarvestStreamWithPagination):
+    @property
+    @abstractmethod
+    def parent_stream_name(self) -> str:
+        """
+        :return: parent stream class name
+        """
+
+    def stream_slices(self, sync_mode: SyncMode, cursor_field: List[str] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        parent_stream = type(self.parent_stream_name, (HarvestStreamWithPagination,), {})
+        items = parent_stream(authenticator=self.authenticator)
+        for item in items.read_records(sync_mode=sync_mode):
+            yield {"parent_id": item["id"]}
+
+
+class Contacts(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Clients(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Company(HarvestStream):
+    primary_key = None
+    data_field = None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield response.json()
+
+    def path(self, **kwargs) -> str:
+        return self.name
+
+
+class Invoices(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class InvoiceMessages(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    parent_stream_name = "Invoices"
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
```

Comment:
lets have `template_path` or something and have default implementation of path in NestedStream class, something like
```python
return self.template_path.format(parent_id=stream_slice['parent_id'])
```
WDYT?
then this class will be one liner